auth/jwtverifier/verifier_test.go

```diff
@@ -0,0 +1,389 @@
+				return req
+			},
+			wantErr:    false,
+			wantClaims: map[string]interface{}{"org": "testorg", "tier": float64(1)},
+		},
+		{
+			name: "missing authorization header",
+			createReq: func() *http.Request {
+				return httptest.NewRequest("GET", "/test", nil)
+			},
+			wantErr: true,
+		},
+		{
+			name: "invalid bearer format",
+			createReq: func() *http.Request {
+				req := httptest.NewRequest("GET", "/test", nil)
+				req.Header.Set("Authorization", "InvalidFormat token")
+				return req
+			},
+			wantErr: true,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			req := tt.createReq()
+			claims, err := verifier.ExtractClaims(req)
+
+			if (err != nil) != tt.wantErr {
+				t.Errorf("ExtractClaims() error = %v, wantErr %v", err, tt.wantErr)
+				return
+			}
+
+			if !tt.wantErr {
+				for key, expectedValue := range tt.wantClaims {
+					if actualValue, ok := claims[key]; !ok {
+						t.Errorf("Missing claim %s", key)
+					} else if actualValue != expectedValue {
+						t.Errorf("Claim %s = %v, want %v", key, actualValue, expectedValue)
+					}
+				}
+			}
+		})
+	}
+}
+
+func TestNewDirectVerifier(t *testing.T) {
+	tests := []struct {
+		name      string
+		jwksURL   string
+		wantErr   bool
+		parseErr  bool // If true, URL parsing itself should fail
+	}{
+		{
+			name:    "valid HTTPS URL",
+			jwksURL: "https://example.com/.well-known/jwks.json",
+			wantErr: false,
+		},
+		{
+			name:    "valid HTTP URL (with warning)",
+			jwksURL: "http://localhost:8080/jwks",
+			wantErr: false,
+		},
+		{
+			name:     "invalid URL",
+			jwksURL:  "://invalid",
+			parseErr: true,
+		},
+		{
+			name:    "invalid scheme",
+			jwksURL: "ftp://example.com/jwks",
+			wantErr: true,
+		},
+		{
+			name:    "nil URL",
+			jwksURL: "",
+			wantErr: true,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			var parsedURL *url.URL
+			var parseErr error
+
+			if tt.jwksURL != "" {
+				parsedURL, parseErr = url.Parse(tt.jwksURL)
+				if tt.parseErr {
+					if parseErr == nil {
+						t.Error("Expected URL parse error, got nil")
+					}
+					return
+				}
+				if parseErr != nil {
+					t.Fatalf("Unexpected URL parse error: %v", parseErr)
+				}
+			}
+
+			verifier, err := NewDirectVerifier(parsedURL)
+			if (err != nil) != tt.wantErr {
+				t.Errorf("NewDirectVerifier() error = %v, wantErr %v", err, tt.wantErr)
+			}
+			if !tt.wantErr && verifier == nil {
+				t.Error("Expected non-nil verifier")
+			}
+		})
+	}
+}
+
+func TestDirectVerifier_Mode(t *testing.T) {
+	testURL, _ := url.Parse("https://example.com/jwks")
+	verifier, _ := NewDirectVerifier(testURL)
+	if mode := verifier.Mode(); mode != "direct" {
+		t.Errorf("Mode() = %v, want direct", mode)
+	}
+}
+
+// Test InsecureVerifier
+func TestInsecureVerifier_ExtractClaims(t *testing.T) {
+	// Set environment variable for insecure mode
+	os.Setenv("ALLOW_INSECURE_JWT", "true")
+	defer os.Unsetenv("ALLOW_INSECURE_JWT")
```

Comment:
🗒️ FYI (and not necessarily related to the correctness of this pull request) `os.Setenv` and `os.Unsetenv` are thread unsafe operations that can lead to memory corruption when used from different thread contexts. Earlier this year, I had this issue in a pet project of mine here: https://github.com/rbmk-project/rbmk/pull/71.